### PR TITLE
dead variable removal + macOS pygame fix

### PR DESCRIPTION
**Pull request by umutucak**
# Description
1. Removed dead variables that were not used, and replaced them with the used variables.
2. Pygame window did not pop up on macOS without the pump() function

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [ ] I have run `pytest -v` and no errors are present.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by elliottower** [comment]
Hey, thanks for the catch with these dead variables. What MacOS version are you currently running? I am on 12.4 and it works as is, maybe you can link some pygame specific information showing the need for `pump`? And is this only the case for multiwalker? If you could test other envs to check that would be great

**Comment by umutucak** [comment]
I am on `Ventura 13.5.2` and `Pygame 2.3.0`.

[Documentation](https://www.pygame.org/docs/ref/event.html#pygame.event.pump) mentions that `pump()` may be needed to communicate the event queue with the OS if other `pygame.event` functions are not being used.

File: pettingzoo/classic/connect_four/connect_four.py [hunk 1]
```diff
@@ -285,6 +285,8 @@ def render(self):
         if self.render_mode == "human":
             pygame.display.update()
             self.clock.tick(self.metadata["render_fps"])
+            pygame.display.flip()
```
Comment (elliottower) [inline pettingzoo/classic/connect_four/connect_four.py:288]: You should only need one of flip and update, and you should check docs and stuff to see but I imagine pomp might be called before these others.

File: pettingzoo/sisl/multiwalker/multiwalker_base.py [hunk 2]
```diff
@@ -692,6 +688,7 @@ def render(self, close=False):
         self.surf = pygame.transform.flip(self.surf, False, True)
         self.screen.blit(self.surf, (-self.scroll * render_scale - offset, 0))
         if self.render_mode == "human":
+            pygame.event.pump()  # macos pygame window popup fix
```
Comment (elliottower) [inline pettingzoo/sisl/multiwalker/multiwalker_base.py:691]: Let’s try to be consistent and always do pump before flip and on the same lines ( with or without comments, ppl can see the git blame if they really want to know why it’s there, but you can leave the comments if you want)

File: pettingzoo/sisl/pursuit/pursuit_base.py [hunk 3]
```diff
@@ -418,6 +418,7 @@ def render(self):
         del observation
         if self.render_mode == "human":
             pygame.display.flip()
+            pygame.event.pump()
```
Comment (elliottower) [inline pettingzoo/sisl/pursuit/pursuit_base.py:421]: Again let’s try to get it consistent with the ordering and such.

**Review by elliottower — APPROVED** [review]
Mostly good just a few comments

**Comment by elliottower** [comment]
@umutucak just tagging you in case you missed the comments. If you aren't able to I can branch from this branch and fix the minor issues myself.

**Comment by umutucak** [comment]
Hi, sorry I've been a bit busy. I will take care of them this week 👍

File: pettingzoo/classic/connect_four/connect_four.py (hunk 1 above)
Comment (umutucak) [reply pettingzoo/classic/connect_four/connect_four.py:288]: I can't find much on the order of `pump` and `update`, but you are correct with not needing both `update` and `flip`

File: pettingzoo/sisl/multiwalker/multiwalker_base.py [hunk 4]
```diff
@@ -688,7 +688,7 @@ def render(self, close=False):
         self.surf = pygame.transform.flip(self.surf, False, True)
         self.screen.blit(self.surf, (-self.scroll * render_scale - offset, 0))
         if self.render_mode == "human":
-            pygame.event.pump()  # macos pygame window popup fix
+            pygame.event.pump()
             pygame.display.flip()
```
[the remark this answers is not in the document]
Comment (umutucak) [reply pettingzoo/sisl/multiwalker/multiwalker_base.py:692]: yes you are right i missed this

**Comment by elliottower** [comment]
Just realized you may not have been fully finished with this… let me know if so and I can revert it (seemed finished to me though)

**Comment by umutucak** [comment]
no, everything is fine for me. thanks a lot for the help 👍

**Pull request merged**